src/transports/smart_pkt.c

```diff
@@ -360,7 +360,15 @@ static int32_t parse_len(const char *line)
 
 	for (i = 0; i < PKT_LEN_SIZE; ++i) {
 		if (!isxdigit(num[i])) {
-			giterr_set(GITERR_NET, "Found invalid hex digit in length");
+			/* Make sure there are no special characters before passing to error message */
+			for (k = 0; k < PKT_LEN_SIZE; ++k) {
+				if(iscntrl(num[k])) {
+					/* Truncate at first control character */
+					num[k] = '\0';
+				}
+			}
+			
+			giterr_set(GITERR_NET, "Invalid digit in hex length: %s", num);
```

Comment:
While we're talking about this, I would like to see single quotes around the output, eg "invalid hex digit in length: '%s'".  (I don't need to see the "found" verb there, but I also admit that I don't have a serious objection.)